packages/commerce-sdk-react/src/hooks/sample.test.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import '@testing-library/jest-dom'
+import {screen, waitFor} from '@testing-library/react'
+import {rest} from 'msw'
+
+import {renderWithProviders, setupMockServer} from '../test-utils'
+import useCommerceApi from './useCommerceApi'
+import {useProduct} from './ShopperProducts'
+import {product} from './ShopperProducts/mock'
+
+const server = setupMockServer(
```

Comment:
Mocking API responses with `msw`. From what I've seen, it works very similar to `nock`. But I chose `msw` because:
- `msw` can run in the browser too (in case we ever need to do that in the future)
- The testing-library recommends using `msw`. For us, we've used it in retail-react-app, while a few pwa-kit-* packages use `nock`.

Reply:
One of the benefits of using Nock was that we could simply run the tests in integration mode so that network requests are sent to our backend and we have more of an end-to-end test. Obviously we don't do this all the time, maybe only on CI at nightly builds. Does MSW have a similar switch feature. 

Nock would also allow us to record responses and save them to be tested against. It looks like we'd be manually creating mocks for all the requests that we  end up making in tests. Is that how it works?

Reply:
> Nock would also allow us to record responses and save them to be tested against. It looks like we'd be manually creating mocks for all the requests that we end up making in tests. Is that how it works?

That's right. Currently there isn't a way to record the responses with `msw`: https://github.com/mswjs/msw/discussions/1060

> One of the benefits of using Nock was that we could simply run the tests in integration mode so that network requests are sent to our backend and we have more of an end-to-end test. Obviously we don't do this all the time, maybe only on CI at nightly builds. Does MSW have a similar switch feature.

I don't see `msw` having this feature either.

All right, let's switch to using `nock`. Can you point me to some examples of our past usage with it? @bendvc